### ISSUE TITLE
Publish (teamsApp/publishAppPackage) fails with a conflict error when the app is in the submitted state

### ISSUE THREAD

**Issue by therealjohn** [OfficeDev/TeamsFx#10068]
**Describe the bug**
The `teamsApp/publishAppPackage` action fails for subsequent updates of the app (after it's been previously published to a tenant catalog) if it's still in the Submitted state.

**To Reproduce**
1. Create an app project and provision, deploy, and publish it.
2. Try to publish it again. 

**Expected behavior**
A warning/error message that the app has already been published and needs to be approved before any updates can be made. 

**Additional context**
If the app is in the submitted state, it does not have the internal app ID assigned yet, so this branching logic causes it to try and publish again, which fails with a conflict error. 

https://github.com/OfficeDev/TeamsFx/blob/eaec9fde21af0a73f15011de6977617f901f7658/packages/fx-core/src/component/driver/teamsApp/publishAppPackage.ts#L106

It's expected that you can't do this, but the tooling should give a more actionable understanding of why its failing. We can report back that the app is already published (submitted), but not approved yet.

**Comment by SLdragon** [comment]
Thank you for your valuable feedback as we are currently on holiday and may not be able to reply promptly. We will get back to you after 10.6 as soon as possible.

**Comment by nliu-ms** [comment]
Thanks @therealjohn , I will take a look and see if we can improve the error message.

**Comment by wh-alice** [comment]
Close stale issues. Please create new issue then we can support.
